### PR TITLE
Shortcode script

### PR DESCRIPTION
**Pull request by cryptidgit**
In this branch I've implemented a script that automatically appends the first and last name to an email to target message when the name fields lose focus. It only runs if there are no name shortcodes present.

### CODE REVIEW

**Review by cryptidgit — COMMENTED** [review]
These are the changes that I've made to address the comments you've made in the first PR review.

File: packages/common/src/add-name-to-message.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
+/*
+ Adds first and last name when First Name and Last Name fields lose focus if name shortcodes aren't present
+*/
+
+import { ENGrid } from ".";
+
+export class addNameToMessage {
```
Comment (fernanDOTdo) [inline packages/common/src/add-name-to-message.ts:7]: For class names, we use PascalCase. Please change it to AddNameToMessage.

File: packages/common/src/add-name-to-message.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    }
+
+    replaceNameShortcode(fName: string, lName: string) {
+        const firstName: HTMLInputElement | null = document.querySelector(fName);
+        const lastName: HTMLInputElement | null = document.querySelector(lName);
+        let message: HTMLTextAreaElement | null = document.querySelector('[name="contact.message"]');
+        let addedFirstName: boolean = false;
+        let addedLastName: boolean = false;
+        
+        if(ENGrid.getPageType() == "EMAILTOTARGET" && message) {
```
Comment (fernanDOTdo) [inline packages/common/src/add-name-to-message.ts:28]: The `ENGrid.getPageType() == "EMAILTOTARGET"` verification was already made on the `shouldRun` function.

File: packages/common/src/add-name-to-message.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+            if(message.value.includes("{user_data~First Name") || message.value.includes("{user_data~Last Name")) {
+                return;
+            } else {
+                if(!message.value.includes("{user_data~First Name") && firstName) {
+                    firstName.addEventListener("blur", function(e: FocusEvent) {
```
Comment (fernanDOTdo) [inline packages/common/src/add-name-to-message.ts:33]: We can use an arrow function here.

File: packages/common/src/add-name-to-message.ts [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+                        if(message && !addedFirstName){ 
+                            addedFirstName = true;
+                            message.value = message.value.concat("\n" + target.value);
+                        }
+                    });
+                }
+                
+                if(!message.value.includes("{user_data~Last Name") && lastName) {
+                    lastName.addEventListener("blur", function(e) {
```
Comment (fernanDOTdo) [inline packages/common/src/add-name-to-message.ts:44]: We can use an arrow function here too.

**Review by fernanDOTdo — CHANGES_REQUESTED** [review]
Thank you, @cryptidgit !

File: packages/common/src/add-name-to-message.ts (hunk 1 above)
Comment (cryptidgit) [reply packages/common/src/add-name-to-message.ts:7]: I've changed the class name.

File: packages/common/src/add-name-to-message.ts (hunk 2 above)
Comment (cryptidgit) [reply packages/common/src/add-name-to-message.ts:28]: I've removed the repeating page type check.

**Review by fernanDOTdo — CHANGES_REQUESTED** [review]
@cryptidgit Everything is ready to merge. Please, sync your branch with the current main so I can merge it.
Thank you!

**Comment by fernanDOTdo** [comment]
@cryptidgit Never mind! I'm gonna do it for you.

**Pull request merged**